tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,358 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: This kubevirt test relies on the upstream test code, which is downstreamed as virt-tests.
+#          This is the part running on server node.
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base multi_machine_job_base;
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use mmapi;
+use utils;
+use File::Basename;
+
+my @kubevirt_tests = (
+    'vmi_lifecycle_test',
+    'vm_test',
+    'vmipreset_test',
+    # 'vmi_configuration_test', # Disable this test, because VMI Huge page settings needs to be done first
+    'vmi_headless_test',
+    'access_test',
+    'storage_test',
+    'container_disk_test',
+    'datavolume_test',
+    'hotplug_test',
+    'vmi_cloudinit_test',
+    'vmi_iothreads_test',
+    'vmi_multiqueue_test',
+    'primary_pod_network_test',
+    'vmi_multus_test',
+    'vmi_networking_test',
+    'networkpolicy_test',
+    'vmi_monitoring_test',
+    'vnc_test',
+    'console_test',
+    'credentials_test',
+    'expose_test',
+    'replicaset_test',
+    'network_services_test',
+    'vmi_kernel_boot_test',
+    'infra_test',
+    'operator_test',
+    'migration_test',
+    'subresource_api_test',
+    'virt_control_plane_test'
+);
+
+sub run {
+    my ($self) = shift;
+    my $sut_ip = get_required_var('SUT_IP');
+
+    set_var('SERVER_IP', $sut_ip);
+    bmwqemu::save_vars();
+
+    # Synchronize the server & agent nodes
+    barrier_create('KUBEVIRT_TEST_SETUP', 2);
+    barrier_wait('KUBEVIRT_TEST_SETUP');
+    # mutex_wait('KUBEVIRT_TEST_SETUP');
+
+    my $agent_ip = $self->get_var_from_parent("AGENT_IP");
+    record_info('Agent IP', $agent_ip);
+
+    record_info('Start RKE2 server setup');
+    $self->rke2_server_setup($agent_ip);
+
+    record_info('Deploy kubevirt manifests');
+    $self->deploy_kubevirt_manifests();
+    
+    record_info('Run kubevirt tests');
+    $self->run_virt_tests();
+    
+    record_info('Generate test report');
+    $self->generate_test_report();
+    
+    record_info('Upload results');
+    $self->upload_results();
+
+    barrier_create('KUBEVIRT_TEST_DONE', 2);
+    barrier_wait('KUBEVIRT_TEST_DONE');
```

Comment:
It seems that both agent and server create KUBEVIRT_TEST_DONE

Reply:
As per the usage introduction https://github.com/os-autoinst/openQA/blob/master/docs/WritingTests.asciidoc#test-synchronization-and-locking-api, it doesn't work from my test. I don't know why.